internal/staticupload/upload.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package staticupload
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	s3manager "github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+// Upload uploads the given object to S3 and invalidates the CDN cache.
+// It returns the upload output or an error.
+// The error will be of type InvalidationError if the CDN cache could not be invalidated.
```

Comment:
Might be a bit too much information, but maybe we should add that this will only happen, if `CacheInvalidationWaitTimeout != 0`, right?

Reply:
That's not a correct statement. If the request to start the invalidation fails, `Upload` will return an `InvalidationError`, regardless of the `CacheInvalidationWaitTimeout`.

Reply:
Ah, I overlooked that.